src/app/CommandSender.cpp

```diff
@@ -296,11 +323,13 @@ CHIP_ERROR CommandSender::ProcessInvokeResponseIB(InvokeResponseIB::Parser & aIn
     StatusIB statusIB;
 
     {
-        bool hasDataResponse = false;
+        bool commandRefExpected = (mFinishedCommandCount > 1);
+        bool hasDataResponse    = false;
         TLV::TLVReader commandDataReader;
 
         CommandStatusIB::Parser commandStatus;
-        err = aInvokeResponse.GetStatus(&commandStatus);
+        err                         = aInvokeResponse.GetStatus(&commandStatus);
+        mAdditionalResponseElements = AdditionalInvokeResponseElements();
```

Comment:
nit: should we move this one line up so that `err` initialization and status check are grouped/close together without any intermediate code in between them?